extension/deployment/src/main/java/io/quarkiverse/temporal/deployment/TemporalProcessor.java

```diff
@@ -254,10 +258,29 @@ void validateMockConfiguration(Capabilities capabilities) {
     @Record(ExecutionTime.RUNTIME_INIT)
     WorkflowClientBuildItem recordWorkflowClient(
             WorkflowServiceStubsRecorder recorder,
-            WorkflowClientRecorder clientRecorder) {
+            WorkflowClientRecorder clientRecorder,
+            TemporalBuildtimeConfig temporalBuildtimeConfig,
+            CombinedIndexBuildItem beanArchiveBuildItem) {
+
+        List<ContextPropagator> propagators = new ArrayList<>();
+        temporalBuildtimeConfig.contextPropagatorClasses().ifPresent(classes -> {
+            for (String workflowClass : classes) {
+                DotName className = DotName.createSimple(workflowClass);
+                ClassInfo classInfo = beanArchiveBuildItem.getIndex().getClassByName(className);
+                if (classInfo.interfaceNames().contains(CONTEXT_PROPAGATOR)) {
+                    throw new ConfigurationException("Class " + workflowClass + " is not a context propagator");
+                }
+                try {
+                    propagators.add((ContextPropagator) Class.forName(workflowClass).getDeclaredConstructor().newInstance());
```

Comment:
Basically I need to call the default constructor on your propagator class not sure if three is a better way to do it than this old fashioned way?

Reply:
could we leverage CDI ? not sure if this a good idea or not to make this a CDI bean

Reply:
but this way we could just retrieve the instance without having to deal with instantiation

Reply:
i guess it could be a CDI bean but we were just going on their exmaple code: https://github.com/Quinn-With-Two-Ns/samples-java/commit/c99e24af5f20d1965827c5e178f6541a9b5ddf12#diff-110913636fdd14fbceea63df5e4a32e6815c72549d7420cea869c3f8455fc886